sync2/storage.go

```diff
@@ -177,8 +186,3 @@ func (s *Storage) UpdateDeviceSince(deviceID, since string) error {
 	_, err := s.db.Exec(`UPDATE syncv3_sync2_devices SET since = $1 WHERE device_id = $2`, since, deviceID)
 	return err
 }
-
-func (s *Storage) UpdateUserIDForDevice(deviceID, userID string) error {
-	_, err := s.db.Exec(`UPDATE syncv3_sync2_devices SET user_id = $1 WHERE device_id = $2`, userID, deviceID)
-	return err
-}
```

Comment:
We now need to know the user_id in order to form a device_id; prior to this it was possible that the user id had not yet been fetched from /whoami.

Come to think of it, we could make the user_id column non null here.

Reply:
Yes please.

Reply:
Ahh, it already is null. It's just that sometimes we put the empty string into this column.

There are six entries in this table with `user_id = ''`. But they all have `since = ''` too. I think we ought to drop such devices from the table.